losoto/h5parm.py

```diff
@@ -61,23 +61,49 @@ class h5parm( object ):
         compression level from 0 to 9 when creating the file, by default 5.
     complib : str, optional
         library for compression: lzo, zlib, bzip2, by default zlib.
+
+    Notes
+    -----
+    An `h5parm` object may be used in a ``with`` context. However, be aware
+    that the underlying HDF5 file object will be closed as soon as you exit
+    the context. To re-open the HDF5 file object, you need to create a new
+    ``with`` context using the current `h5parm` object.
+    Calling `h5parm.open()` explicitly is *not* recommended, because it is
+    not exception-safe.
     """
 
     def __init__(self, h5parmFile, readonly=True, complevel=0, complib='zlib'):
-
+        """
+        Initialize `h5parm` object.
+        """
         self.H = None  # variable to store the pytable object
         self.fileName = h5parmFile
 
-        if os.path.isfile(h5parmFile):
-            if not tables.is_hdf5_file(h5parmFile):
-                logging.critical('Not a HDF5 file: '+h5parmFile+'.')
-                raise Exception('Not a HDF5 file: '+h5parmFile+'.')
-            if readonly:
-                logging.debug('Reading from '+h5parmFile+'.')
-                self.H = tables.open_file(h5parmFile, 'r', IO_BUFFER_SIZE=1024*1024*10, BUFFER_TIMES=500)
+        self._readonly = readonly
+        self._complevel = complevel
+        self._complib = complib
+
+        self.open()
+
+
+    def open(self):
+        """
+        Create and open the HDF5 file object, unless it already exists and is
+        already opened.
+        """
+        if self.H and self.H.isopen:
+            return
+
+        if os.path.isfile(self.fileName):
+            if not tables.is_hdf5_file(self.fileName):
+                logging.critical('Not a HDF5 file: '+self.fileName+'.')
```

Comment:
I guess this logging call can be removed.

Reply:
> I guess this logging call can be removed.

It was there already. I wanted to change as little as possible, but I agree with you. I'll remove it.